csharp/private/rules/binary.bzl

```diff
@@ -10,6 +10,8 @@ load(
 def _binary_impl(ctx):
     providers = {}
 
+    stdlib = [ctx.attr.stdlib] if ctx.attr.stdlib else []
```

Comment:
Soon it'd be reasonable to de-dupe the code in these rules... I've been putting it off until we support more stuff because I'm not sure exactly what the API should look like.

`AssemblyAction` by contrast I've been using as an interface for csc, so it's API (very) roughly matches it.